engines/wage/design.cpp

```diff
@@ -0,0 +1,539 @@
+				Patterns &patterns, byte fillType, byte borderThickness, byte borderFillType) {
+	int16 y1 = in.readSint16BE();
+	int16 x1 = in.readSint16BE();
+	int16 y2 = in.readSint16BE();
+	int16 x2 = in.readSint16BE();
+
+	if (x1 > x2)
+		SWAP(x1, x2);
+	if (y1 > y2)
+		SWAP(y1, y2);
+
+	Common::Rect r(x1, y1, x2, y2);
+	plotData pd(surface, &patterns, fillType, 1);
+
+	if (fillType <= patterns.size())
+		Graphics::drawFilledRect(r, kColorBlack, drawPixel, &pd);
+
+	pd.fillType = borderFillType;
+	pd.thickness = borderThickness;
+
+	if (borderThickness > 0 && borderFillType <= patterns.size()) {
+		Graphics::drawLine(x1, y1, x2, y1, kColorBlack, drawPixel, &pd);
+		Graphics::drawLine(x2, y1, x2, y2, kColorBlack, drawPixel, &pd);
+		Graphics::drawLine(x2, y2, x1, y2, kColorBlack, drawPixel, &pd);
+		Graphics::drawLine(x1, y2, x1, y1, kColorBlack, drawPixel, &pd);
+	}
+}
+
+void Design::drawRoundRect(Graphics::Surface *surface, Common::ReadStream &in,
+				Patterns &patterns, byte fillType, byte borderThickness, byte borderFillType) {
+	int16 y1 = in.readSint16BE();
+	int16 x1 = in.readSint16BE();
+	int16 y2 = in.readSint16BE();
+	int16 x2 = in.readSint16BE();
+	int16 arc = in.readSint16BE();
+
+	if (x1 > x2)
+		SWAP(x1, x2);
+	if (y1 > y2)
+		SWAP(y1, y2);
+
+	Common::Rect r(x1, y1, x2, y2);
+	plotData pd(surface, &patterns, fillType, 1);
+
+	if (fillType <= patterns.size())
+		Graphics::drawRoundRect(r, arc/2, kColorBlack, true, drawPixel, &pd);
+
+	pd.fillType = borderFillType;
+	pd.thickness = borderThickness;
+
+	if (borderThickness > 0 && borderFillType <= patterns.size())
+		Graphics::drawRoundRect(r, arc/2, kColorBlack, false, drawPixel, &pd);
+}
+
+void Design::drawPolygon(Graphics::Surface *surface, Common::ReadStream &in,
+	Patterns &patterns, byte fillType, byte borderThickness, byte borderFillType) {
+
+	byte ignored = in.readSint16BE(); // ignored
+	assert(ignored == 0);
+
+	int numBytes = in.readSint16BE(); // #bytes used by polygon data, including the numBytes
+	int16 by1 = in.readSint16BE();
+	int16 bx1 = in.readSint16BE();
+	int16 by2 = in.readSint16BE();
+	int16 bx2 = in.readSint16BE();
+	Common::Rect bbox(bx1, by1, bx2, by2);
+
+	numBytes -= 8;
+
+	int y1 = in.readSint16BE();
+	int x1 = in.readSint16BE();
+
+	Common::Array<int> xcoords;
+	Common::Array<int> ycoords;
+
+	numBytes -= 6;
+
+	while (numBytes > 0) {
+		int y2 = y1;
+		int x2 = x1;
+		int b = in.readSByte();
+		if ((b & 0xff) == 0x80) {
+			y2 = in.readSint16BE();
+			numBytes -= 3;
+		} else {
+			y2 += b;
+			numBytes -= 1;
+		}
+		b = in.readSByte();
+		if ((b & 0xff) == 0x80) {
+			x2 = in.readSint16BE();
+			numBytes -= 3;
+		} else {
+			x2 += b;
+			numBytes -= 1;
+		}
+		xcoords.push_back(x1);
+		ycoords.push_back(y1);
+		x1 = x2;
+		y1 = y2;
+	}
+	xcoords.push_back(x1);
+	ycoords.push_back(y1);
+
+	int npoints = xcoords.size();
+	int *xpoints = (int *)calloc(npoints, sizeof(int));
+	int *ypoints = (int *)calloc(npoints, sizeof(int));
```

Comment:
You can easily do `xcoords.begin()` and `ycoords.begin()` to pass them to `Graphics::drawPolygonScan`